samples/apps/copilot-chat-app/importdocument/Program.cs

```diff
@@ -67,14 +68,15 @@ public static void Main(string[] args)
                 .WithRedirectUri(config.RedirectUri)
                 .Build();
             var result = await app.AcquireTokenInteractive(scopes).ExecuteAsync();
-            var accounts = await app.GetAccountsAsync();
-            if (!accounts.Any())
+            IEnumerable<IAccount>? accounts = await app.GetAccountsAsync();
+            IEnumerable<IAccount> accountList = accounts.ToList();
+            if (!accountList.Any())
             {
                 Console.WriteLine("Error: No accounts found");
                 return null;
             }
 
-            return accounts.First().HomeAccountId.Identifier;
+            return accountList.First().HomeAccountId.Identifier;
```

Comment:
This actually makes it more expensive, as it's forcing a full enumeration of the accounts in order to turn it into a `List<T>`, allocating a `List<T>` and enough space to hold the results, etc.  In contrast, at worst previously `Any` would allocate an enumerator and just look at the first element, but if the `IEnumerable<T>` returned from `GetAccountsAsync` was actually an `ICollection<T>`, then `Any` will just read its Count property.  Similarly, `First` won't enumerate more than a single item.

Since it looks like the implementation doesn't allow nor account for null IAccounts, you can do this whole thing with just a single FirstOrDefault call:
```C#
IEnumerable<IAccount> accounts = await app.GetAccountsAsync();
IAccount? first = accounts.FirstOrDefault();
if (first is null)
{
    Console.WriteLine("Error: No accounts found");
    return null;
}

return first.HomeAccountId.Identifier;
```

Reply:
good catch, thanks :-)